EXOSIMS/Observatory/TwoStarShades_mission.py

```diff
@@ -0,0 +1,567 @@
+            },
+            tol=Tol,
+            options={"disp": False},
+        )
+        print(res)
+        print(res.dtype())
+        print(res.x)
+        breakpoint()
+        # opt_slewTime = res.x
+        # opt_dV = self.calculate_dV(opt_slewTime, TL, nA, nB, tA)
+        # calculating in correct order
+        opt_slewTime = res.x * u.d
+
+        opt_dV = self.calculate_dV(TL, nA, nB, sd, opt_slewTime, tA)
+
+        return opt_slewTime, opt_dV.value
+
+    def log_occulterResults(self, DRM, slewTimes, sInd, sd, dV):
+        """Updates the given DRM to include occulter values and results
+
+        Args:
+            DRM (dict):
+                Design Reference Mission, contains the results of one complete
+                observation (detection and characterization)
+            slewTimes (astropy.units.Quantity):
+                Time to transfer to new star line of sight in units of days
+            sInd (int):
+                Integer index of the star of interest
+            sd (astropy.units.Quantity):
+                Angular separation between stars in rad
+            dV (astropy.units.Quantity):
+                Delta-V used to transfer to new star line of sight in units of m/s
+
+        Returns:
+            dict:
+                Design Reference Mission dictionary, contains the results of one
+                complete observation (detection and characterization)
+
+        """
+        if self.counter_3 == 0:
+            DRM["slew_time_1"] = slewTimes.to("day")
+            DRM["slew_angle_1"] = sd.to("deg")
+
+            slew_mass_used = slewTimes * self.defburnPortion * self.flowRate
+            DRM["slew_dV_1"] = (slewTimes * self.ao[:, 0] * self.defburnPortion).to(
+                "m/s"
+            )
+            DRM["slew_mass_used_1"] = slew_mass_used.to("kg")
+            self.scMass[:, 0] = self.scMass[:, 0] - slew_mass_used
+            DRM["scMass_1"] = self.scMass[:, 0].to("kg")
+            if self.twotanks:
+                self.slewMass = self.slewMass - slew_mass_used
+                DRM["slewMass"] = self.slewMass.to("kg")
+            self.counter_3 = self.counter_3 + 1
+            return DRM
+        else:
+            DRM["slew_time_2"] = slewTimes.to("day")
+            DRM["slew_angle_2"] = sd.to("deg")
+
+            slew_mass_used_2 = slewTimes * self.defburnPortion * self.flowRate
+            DRM["slew_dV_2"] = (slewTimes * self.ao[:, 1] * self.defburnPortion).to(
+                "m/s"
+            )
+            DRM["slew_mass_used_2"] = slew_mass_used_2.to("kg")
+            self.scMass[:, 1] = self.scMass[:, 1] - slew_mass_used_2
+            DRM["scMass_2"] = self.scMass[:, 1].to("kg")
+            if self.twotanks:
+                self.slewMass = self.slewMass - slew_mass_used
+                DRM["slewMass"] = self.slewMass.to("kg")
+            self.counter_3 = 0
+            return DRM
+
+    def refuel_tank(self, TK, tank=None):
```

Comment:
Note that there was a logical error in the original refuel_tank implementation.  See the recent changes in v3.1.6 to see how you should fix your copied version.

Reply:
fixed